picard/ui/scripteditor.py

```diff
@@ -383,14 +509,44 @@ def synchronize_selected_example_lines(current_row, source, target):
     def match_after_to_before(self):
         """Sets the selected item in the 'after' list to the corresponding item in the 'before' list.
         """
-        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_before, self.ui.example_filename_after)
+        self.synchronize_selected_example_lines(self.examples_current_row, self.ui.example_filename_before, self.ui.example_filename_after)
 
     def match_before_to_after(self):
         """Sets the selected item in the 'before' list to the corresponding item in the 'after' list.
         """
-        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+        self.synchronize_selected_example_lines(self.examples_current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+
+    def delete_script(self):
+        """Removes the currently selected script from the script selection combo box and script list.
+        """
+        if self.confirmation_dialog(_('Are you sure that you want to delete the script?')):
+            idx = self.ui.preset_naming_scripts.currentIndex()
+            self.ui.preset_naming_scripts.blockSignals(True)
+            self.ui.preset_naming_scripts.removeItem(idx)
+            self.ui.preset_naming_scripts.setCurrentIndex(0)
```

Comment:
When deleting it is a bit weird it goes to first element after deletion, perhaps it should select the next element instead (or previous one if none). Dunno exactly what's best in this case, but jumping to first element is disturbing.